man/man3/World.3
.TH "World" 3 "Version 0.2.6.0" "EEGGaming And Blinkbird" \" -*- nroff -*-
.ad l
.nh
.SH NAME
World
.SH SYNOPSIS
.br
.PP
.PP
Inherits Node2D\&.
.SS "Public Member Functions"

.in +1c
.ti -1c
.RI "override void \fB_Ready\fP ()"
.br
.ti -1c
.RI "override void \fB_Process\fP (double delta)"
.br
.in -1c
.SS "Static Public Attributes"

.in +1c
.ti -1c
.RI "const float \fBRAISESCOREBY\fP = (float)(1)"
.br
.ti -1c
.RI "static double \fBScore\fP = 0"
.br
.ti -1c
.RI "static \fBGamingSesion\fP \fBgamingSesion\fP = new \fBGamingSesion\fP()"
.br
.ti -1c
.RI "static Timer \fBtimer\fP"
.br
.ti -1c
.RI "static \fBMountains\fP \fBMountains\fP"
.br
.ti -1c
.RI "static \fBPlayer\fP \fBplayer\fP"
.br
.ti -1c
.RI "static Label \fBspeedvalue\fP"
.br
.ti -1c
.RI "static HSlider \fBHslSpeed\fP"
.br
.ti -1c
.RI "static \fBCamera\fP \fBCamera\fP"
.br
.in -1c
.SH "Detailed Description"
.PP 
Definition at line \fB8\fP of file \fBWorld\&.cs\fP\&.
.SH "Member Function Documentation"
.PP 
.SS "override void World\&._Process (double delta)"

.PP
Definition at line \fB116\fP of file \fBWorld\&.cs\fP\&.
.SS "override void World\&._Ready ()"

.PP
Definition at line \fB32\fP of file \fBWorld\&.cs\fP\&.
.SH "Member Data Documentation"
.PP 
.SS "\fBCamera\fP World\&.Camera\fR [static]\fP"

.PP
Definition at line \fB24\fP of file \fBWorld\&.cs\fP\&.
.SS "\fBGamingSesion\fP World\&.gamingSesion = new \fBGamingSesion\fP()\fR [static]\fP"

.PP
Definition at line \fB17\fP of file \fBWorld\&.cs\fP\&.
.SS "HSlider World\&.HslSpeed\fR [static]\fP"

.PP
Definition at line \fB23\fP of file \fBWorld\&.cs\fP\&.
.SS "\fBMountains\fP World\&.Mountains\fR [static]\fP"

.PP
Definition at line \fB19\fP of file \fBWorld\&.cs\fP\&.
.SS "\fBPlayer\fP World\&.player\fR [static]\fP"

.PP
Definition at line \fB20\fP of file \fBWorld\&.cs\fP\&.
.SS "const float World\&.RAISESCOREBY = (float)(1)\fR [static]\fP"

.PP
Definition at line \fB14\fP of file \fBWorld\&.cs\fP\&.
.SS "double World\&.Score = 0\fR [static]\fP"

.PP
Definition at line \fB15\fP of file \fBWorld\&.cs\fP\&.
.SS "Label World\&.speedvalue\fR [static]\fP"

.PP
Definition at line \fB22\fP of file \fBWorld\&.cs\fP\&.
.SS "Timer World\&.timer\fR [static]\fP"

.PP
Definition at line \fB18\fP of file \fBWorld\&.cs\fP\&.

.SH "Author"
.PP 
Generated automatically by Doxygen for EEGGaming And Blinkbird from the source code\&.


BlinkBird/Code/World.cs
using EEGGaming.Core.Data.Models;
using EEGGaming.Core.Managers;
using Godot;
using ScottPlot.AxisRules;
using System;
using System.Drawing;
/// <summary>
/// The class representation for the gaming world 
/// </summary>
public partial class World : Node2D
{
	// Called when the node enters the scene tree for the first time.
	 PackedScene packedScene;
	DateTime started = new DateTime();
	Label lblElapsed, lblScore;
	public const float RAISESCOREBY = (float)(1);
	public static double Score = 0;
	
	public static GamingSesion gamingSesion = new GamingSesion();
	public static Timer  timer;
    public static Mountains Mountains;
	public static Player player;
    //Camera2D camera;
    public static Label speedvalue;
    public static HSlider HslSpeed;
	public static Camera Camera;

    /// <summary>
	/// Changes the speed to the horizontalslider's value
	/// </summary>
	/// <param name="value"></param>
    private void HSdrHorizontalSpeed_ValueChanged(double value)
    {
         speedvalue.Text = Convert.ToString(HslSpeed.Value);
    }

	
    public override void _Ready()
	{
        Camera = (Camera)this.FindChild("Camera",true);
        packedScene = (PackedScene)GD.Load("res://Scenes/Mountains.tscn");
		TextureRect background = (TextureRect)this.FindChild("Background", true);
         
        player =(Player) this.FindChild("Player", true);
        HslSpeed = (HSlider)this.FindChild("HSdrHorizontalSpeed",true);
        HslSpeed.ValueChanged += HSdrHorizontalSpeed_ValueChanged;
        speedvalue = GetNode<Label>("/root/World/HSdrHorizontalSpeed/lblSpeedPercent");

		GD.Print(Camera.Name);

        if (background != null  )
		{

			Vector2 winsize = this.GetWindow().Size;
			winsize.Y = winsize.Y + 250;
			background.Size = winsize;
             

        }
		if ( Menu.ignoreUserselectionandconstateofdevice)
		{
            started = DateTime.Now;

            timer = (Timer)this.FindChild("TmrTimer", true);
            timer.Timeout += Timer_Timeout;
            timer.Start();
            lblElapsed = (Label)this.FindChild("lblElapsed", true);
            lblScore = (Label)this.FindChild("lblScore", true);
        }
		
		if (Menu.recordManager.Sensor != null &&
			Menu.recordManager.Sensor.State== NeuroSDK.SensorState.StateInRange)
		{
			started = DateTime.Now;
			
			timer = (Timer)this.FindChild("TmrTimer", true);
			timer.Timeout += Timer_Timeout;
			timer.Start();
			lblElapsed = (Label)this.FindChild("lblElapsed", true);
			lblScore = (Label)this.FindChild("lblScore", true);


			

			gamingSesion.Start = DateTime.Now
	;
			gamingSesion.User = Menu.SelectedUser.Id;
			gamingSesion.Score = 0;
			gamingSesion = Menu.gamingSesionManager.AddGamingSession(gamingSesion);
			Menu.recordManager.ActiveGamingSessionId = gamingSesion.Id;
			

            Menu.recordManager.ActiveUserId = Menu.SelectedUser.Id;
            GD.Print("SESS ID : " + Menu.recordManager.ActiveGamingSessionId);

            GD.Print("USER ID: " + Menu.recordManager.ActiveUserId);
            
			Menu.recordManager.UsesDb = true;
            Menu.recordManager.Start();
           

        }
        Mountains = (Mountains)packedScene.Instantiate();
       // GD.Print("MOUNTINS ADDED");
        this.AddChild(Mountains);
		Mountains.ZIndex = 2;
		speedvalue.Text = Convert.ToString(HslSpeed.Value);






    }
     
	/// <summary>
	/// Changes the time that elapsed
	/// </summary>
    private void Timer_Timeout()
	{
		if (lblElapsed != null)
		{
			TimeSpan timediff = DateTime.Now.Subtract(started);
			lblElapsed.Text = String.Format("{0} : {1} : {2} : {3}",
				timediff.Hours, timediff.Minutes,timediff.Seconds,
				timediff.Milliseconds);
			lblScore.Text = Convert.ToString(World.Score);



		}
		 
	}

	// Called every frame. 'delta' is the elapsed time since the previous frame.
	public override void _Process(double delta)
	{
		
		




	}
	}


